### PR TITLE
gh-102029: Deprecate passing arguments to `_PyRLock` in `threading`

### PR DESCRIPTION
**Pull request by sobolevn**
I hope that this is not really a breaking change. Here's my reasoning:
1. `_PyRLock` was the first one (since python2). People are familiar with it: and its signature was always correct
2. Since passing values never made any sense, I hope nobody does rely on it
3. Subclasses still work correctly, eventhough they are not really recommened (and `RLock` is exposed as a function)
4. `typeshed` always had `RLock` defined as `() -> RLock` with no complains: https://github.com/python/typeshed/blob/40d853cbe239bcf1479e25c2c59615158b672610/stdlib/threading.pyi#L115-L122
5. I was not able to find any references to `RLock` with arguments

As the next step, I will propose to deprecate the Python version.

* Issue: gh-102029

### CODE REVIEW

**Comment by sobolevn** [comment]
@pablogsal friendly ping :)

**Review by gpshead — COMMENTED** [review]
This makes sense to me, but I want to run it through a huge pile of tests at work just to see what - if any - oddball code was relying on passing ignored values. I hope nothing. I'll do that this week.

**Comment by sobolevn** [comment]
Thank you! In the meantime, I will change the `What's new` entry.

**Comment by gpshead** [comment]
note to self: I included this PR in an internal test run today, I'll have results by tonight/tomorrow.

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -97,6 +97,15 @@ pathlib
   :meth:`~pathlib.Path.rglob`.
   (Contributed by Barney Gale in :gh:`77609`.)
 
+threading
+---------
+
+* Passing any arguments to :func:`threading.RLock` is now restricted.
+  Previously C version allowed any numbers of args and kwargs,
+  but they were just ignored. Python version never allowed any arguments.
```
Comment (gpshead) [inline Doc/whatsnew/3.13.rst]: This statement isn't accurate; Python <= 2.7 and 3.1 accepted a single undocumented `verbose` argument.

File: Lib/threading.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def gettrace():
 
 Lock = _allocate_lock
 
-def RLock(*args, **kwargs):
+def RLock():
```
Comment (gpshead) [inline Lib/threading.py]: Lets do a regular DeprecationWarning API change cycle on this.

```
if args or kwargs:
    warnings.warn(DeprecationWarning, ...)
```

we can put that in 3.12 remove the ignored args/kwargs in 3.14.

**Review by gpshead — CHANGES_REQUESTED** [review]

File: Lib/threading.py (hunk 2 above)
Comment (sobolevn) [reply Lib/threading.py]: I thought that 3.12 is now only accepting bugfixes, isn't it? I would prefer to have this change as safe as possible and to put it in 3.13+. Unless you insist :)

**Comment by sobolevn** [comment]
I have made the requested changes; please review again

File: Doc/whatsnew/3.13.rst [hunk 3]
```diff
@@ -106,6 +106,7 @@ pathlib
   :meth:`~pathlib.Path.rglob`.
   (Contributed by Barney Gale in :gh:`77609`.)
 
+
 Optimizations
```
Comment (sobolevn) [inline Doc/whatsnew/3.13.rst]:
```suggestion
Optimizations
```

**Comment by sobolevn** [comment]
@gpshead now this is deprecated in 3.13 and stated to be changed in 3.15 :)

**Comment by ambv** [comment]
Closing and re-opening to retrigger CLA checks. Sorry for the noise.

**Review by gpshead — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks everyone! CI is now green :)

**Comment by sobolevn** [comment]
Can someone please click "merge"? :)

**Pull request merged**